src/serialize/column.rs

```diff
@@ -291,14 +297,21 @@ where
     }
 
     let predicate = |x: &&Archetype| -> bool { !x.is_empty() && x.satisfies::<Q>() };
-    let mut seq = serializer.serialize_seq(Some(world.archetypes().filter(predicate).count()))?;
+    let mut seq = serializer.serialize_seq(Some(world.archetypes().filter(predicate).count() + 2))?;
+    
+    let generations: Vec<u32> = world.entities_meta().iter().map(|m| m.generation.get()).collect();
```

Comment:
Serialization logic must not use non-public API. Anything these modules do must be achievable by external code, because users may need to serialize a world using tools other than serde.

Reply:
Fixed.